src/ros_filter.cpp

```diff
@@ -46,6 +48,26 @@
 
 namespace RobotLocalization
 {
+
+  //! @brief Compute covariance volume
+  //!
+  //! Compute the volume of a covariance matrix as its determinant, which is
+  //! computed as the product of the matrix eigenvalues.
+  //!
+  //! If the determinant cannot be computed, the volume returned is +Inf.
+  //!
+  //! @return Volume
+  //!
+  template <typename T, int N>
+  T computeCovarianceVolume(const Eigen::Matrix<T, N, N>& M)
+  {
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<T, N, N> > eigensolver(M);
+
+    return eigensolver.info() == Eigen::Success ?
+           eigensolver.eigenvalues().prod() :
+           std::numeric_limits<T>::infinity();
+  }
+
```

Comment:
Do we want to include the update vector here? We may not want to warn them about variables that they aren't using, correct? For example, if they have Z set to false in their configuration, then we should ignore any changes in its covariance, no? We would have to compute a sub-matrix (either in here or when we call it) that contains only the variables in question.